File: gerador_pontos.py
import random
import os

quantidades_pontos = [1000, 10000, 100000, 1000000]

min_coord = 0.0
max_coord = 100.0

casas_decimais = 2

print(f"Iniciando a geração dos arquivos TXT com pontos aleatórios (intervalo: [{min_coord}, {max_coord}])...")

for N in quantidades_pontos:
    dir = "arquivos_dados"
    if not os.path.exists(dir):
        os.makedirs(dir)

    nome_arquivo = f"pontos_{N}.txt"
    path_arquivo = os.path.join(dir, nome_arquivo)
    
    with open(path_arquivo, 'w') as arquivo:
        for _ in range(N):
            x = random.uniform(min_coord, max_coord)
            y = random.uniform(min_coord, max_coord)
            
            formato = f"%.{casas_decimais}f %." + f"{casas_decimais}f\n"
            linha = formato % (x, y)
            
            arquivo.write(linha)
    
    print(f"Arquivo '{nome_arquivo}' criado com sucesso, contendo {N:,} pontos.")

print("\nProcesso concluído.")

File: gerador_pontos.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
#include <random>
#include <filesystem>
#include <vector>

namespace fs = std::filesystem;

int main()
{
    std::vector<long long> quantidades_pontos = {
        1000,
        10000,
        100000,
        1000000,
        2000000,
    };

    double min_coord = 0.0;
    double max_coord = 100.0;
    int casas_decimais = 2;

    std::cout << "Iniciando a geracao dos arquivos TXT com pontos aleatorios (intervalo: ["
              << min_coord << ", " << max_coord << "])..." << std::endl;

    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_real_distribution<> dis(min_coord, max_coord);

    std::string dir = "arquivos_dados";
    if (!fs::exists(dir))
    {
        fs::create_directory(dir);
    }

    for (auto N : quantidades_pontos)
    {
        std::string nome_arquivo = "pontos_" + std::to_string(N) + ".txt";
        std::string path_arquivo = dir + "/" + nome_arquivo;

        std::ofstream arquivo(path_arquivo);
        if (!arquivo.is_open())
        {
            std::cerr << "Erro ao abrir o arquivo: " << path_arquivo << std::endl;
            continue;
        }

        for (long long i = 0; i < N; i++)
        {
            double x = dis(gen);
            double y = dis(gen);
            arquivo << std::fixed << std::setprecision(casas_decimais)
                    << x << " " << y << "\n";
        }

        arquivo.close();
        std::cout << "Arquivo '" << nome_arquivo << "' criado com sucesso, contendo "
                  << N << " pontos." << std::endl;
    }

    std::cout << "\nProcesso concluido." << std::endl;
    return 0;
}
